qualtran/bloqs/factoring/ecc/_ec_add_r.py

```diff
@@ -0,0 +1,166 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from functools import cached_property
+from typing import Dict
+
+import sympy
+from attrs import frozen
+
+from qualtran import Bloq, bloq_example, BloqDocSpec, QBit, QUInt, Register, Signature, Soquet
+from qualtran.drawing import Circle, TextBox, WireSymbol
+from qualtran.simulation.classical_sim import ClassicalValT
+
+from ._ec_point import ECPoint
+
+
+@frozen
+class ECAddR(Bloq):
+    r"""Perform elliptic curve addition of constant `R`.
+
+    Given the constant elliptic curve point $R$ and an input point $A$
+    factored into the `x` and `y` registers such that $|A\rangle = |(a_x,a_y)\rangle$,
+    this bloq takes
+
+    $$
+    |A\rangle \rightarrow |A+R\rangle
+    $$
+
+    Args:
+        n: The bitsize of the two registers storing the elliptic curve point.
+        R: The elliptic curve point to add.
+
+    Registers:
+        ctrl: A control bit.
+        x: The x component of the input elliptic curve point of bitsize `n`.
+        y: The y component of the input elliptic curve point of bitsize `n`.
+
+    References:
+        [How to compute a 256-bit elliptic curve private key with only 50 million Toffoli gates](https://arxiv.org/abs/2306.08585).
+        Litinski. 2023. Section 1, eq. (3) and (4).
+
+        [Quantum resource estimates for computing elliptic curve discrete logarithms](https://arxiv.org/abs/1706.06752).
+        Roetteler et. al. 2017. Algorithm 1 and Figure 10.
+
+        [https://github.com/microsoft/QuantumEllipticCurves/blob/dbf4836afaf7a9fab813cbc0970e65af85a6f93a/MicrosoftQuantumCrypto/EllipticCurves.qs#L456](QuantumQuantumCrypto).
+        `DistinctEllipticCurveClassicalPointAddition`.
+
+    """
+
+    n: int
+    R: ECPoint
```

Comment:
Are we allowing a departure from pep8 naming convention?

Reply:
hmm I think in this case it probably helps more than it hurts. Something like `ec_point` is unnecessarily long and `r` doesn't match the convention in the paper(s) this module follows. Happy to revisit / change later when more of the bloqs have landed